pkg/virt-controller/watch/snapshot/restore.go

```diff
@@ -514,15 +540,29 @@ func (t *vmRestoreTarget) Reconcile() (bool, error) {
 	}
 	newVM.Annotations[lastRestoreAnnotation] = restoreID
 
-	_, err = t.controller.Client.VirtualMachine(newVM.Namespace).Update(newVM)
+	newVM, err = patchVM(newVM, t.vmRestore.Spec.Patches)
+	if err != nil {
+		return false, fmt.Errorf("error patching VM %s: %v", newVM.Name, err)
+	}
+
+	if !t.doesTargetVMExist() {
+		newVM, err = t.controller.Client.VirtualMachine(t.vmRestore.Namespace).Create(newVM)
+	} else {
+		newVM, err = t.controller.Client.VirtualMachine(newVM.Namespace).Update(newVM)
+	}
 	if err != nil {
 		return false, err
 	}
+	t.vm = newVM
 
 	return true, nil
 }
 
 func (t *vmRestoreTarget) Own(obj metav1.Object) {
+	if !t.doesTargetVMExist() {
```

Comment:
just making sure, will the Own happen after the new VM will be created? cause if not we do want the vm to own the pvcs for example

Reply:
In my brief testing, it appears that the restore will be owned by the target VM but we may want to dig into when/how that happens and if there are any issues because adding the owner reference is followed by setting progressing/ready conditions with "Initializing VirtualMachineRestore"

Reply:
@mhenriks Yes, you're right.

I've just added this test to the functional test I've made:
```go
By("Expecting the restore to be owned by target VM")
	isOwnedByTargetVM := false
	for _, ownReference := range restore.ObjectMeta.OwnerReferences {
		if ownReference.Kind == "VirtualMachine" && ownReference.UID == newVM.UID &&
		   ownReference.Name == newVM.Name {
				isOwnedByTargetVM = true
				break
		}
	}
	Expect(isOwnedByTargetVM).To(BeTrue(), "restore is expected to be owned by target VM")
```

This test had passed and now added to the functional test